src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/PreferencesController.php

```diff
@@ -102,32 +80,53 @@ public function processFormAction(Request $request)
         $form = $this->get('prestashop.adapter.preferences.form_handler')->getForm();
         $form->handleRequest($request);
 
-        if (!$form->isSubmitted()) {
-            return $this->redirectToRoute('admin_preferences');
-        }
+        if ($form->isSubmitted() && $form->isValid()) {
+            $data = $form->getData();
+            $saveErrors = $this->get('prestashop.adapter.preferences.form_handler')->save($data);
 
-        $data = $form->getData();
-        $saveErrors = $this->get('prestashop.adapter.preferences.form_handler')->save($data);
+            if (0 === count($saveErrors)) {
+                /** @var EntityManager $em */
+                $em = $this->get('doctrine.orm.entity_manager');
 
-        if (0 === count($saveErrors)) {
-            /** @var EntityManager $em */
-            $em = $this->get('doctrine.orm.entity_manager');
+                /** @var TabRepository $tabRepository */
+                $tabRepository = $em->getRepository(Tab::class);
 
-            /** @var TabRepository $tabRepository */
-            $tabRepository = $em->getRepository(Tab::class);
+                $tabRepository->changeStatusByClassName(
```

Comment:
I'm wondering if it shouldn't be handled by a CQRS command? But that's probably out of scope 🤔

Reply:
Good point. IMO since I already started improving controllers this can be done as well.

Reply:
I updated the PR to use the command